### PR TITLE
added oauth gateway

### PR DESCRIPTION
**Pull request by acrosdale**
Hello, I would like to make Gametime a better place by contributing the following code:

### Feature/bug description:

1. added oauth gateway on apiv6 + all required func to handle external merchant token
2. added payment method grant

### JIRA link:
https://gametime.atlassian.net/browse/BE-11621

### What can this break:
braintree (hopefully not)

### How has this been tested:
integration testing

tested refreshing token
Screenshot 2024-06-17 at 9 17 16 PM

tested grant (using the access token from test 1 above)
Screenshot 2024-06-17 at 9 50 41 PM

### Sources
https://developer.paypal.com/braintree/docs/guides/extend/oauth/access-tokens/python
https://github.com/braintree/braintree_python/tree/master

### CODE REVIEW

**Review by husobee-gt — APPROVED** [review]

File: braintree.go [hunk 1]
```diff
@@ -56,6 +57,17 @@ func NewWithHttpClient(env Environment, merchantId, publicKey, privateKey string
 	return &Braintree{credentials: newAPIKey(env, merchantId, publicKey, privateKey), HttpClient: client}
 }
 
+// NewV2 creates a Braintree client with API Keys and client credentials.
+// some endpoints require client credentials to be passed in the header.
+func NewV2(env Environment, merchId, pubKey, privKey, clientId, clientSecrets string) *Braintree {
```
Comment (ltaziri) [inline braintree.go:62]: would prefer a better name for this
NewWithCredentials

File: braintree.go [hunk 2]
```diff
@@ -56,6 +57,17 @@ func NewWithHttpClient(env Environment, merchantId, publicKey, privateKey string
...
+	return NewWithHttpClientV2(env, merchId, pubKey, privKey, clientId, clientSecrets, defaultClient)
+}
+
+// NewWithHttpClient creates a Braintree client with API Keys and a HTTP Client.
+func NewWithHttpClientV2(env Environment, merchantId, publicKey, privateKey, clientId, clientSecret string, client *http.Client) *Braintree {
```
Comment (ltaziri) [inline braintree.go:67]: NewHttpClientWithCredentials

File: credentials_api_key.go [hunk 3]
```diff
@@ -18,6 +20,19 @@ func newAPIKey(env Environment, merchantID, publicKey, privateKey string) creden
 	}
 }
 
+func newAPIKeyV2(env Environment, merchantID, publicKey, privateKey, clientId, clientSecret string) credentials {
```
Comment (ltaziri) [inline credentials_api_key.go:23]: newAPIKeyWithCredentials

**Review by ltaziri — COMMENTED** [review]
where are the tests that are in the PR description?

**Comment by acrosdale** [comment]
> where are the tests that are in the PR description?

I tested the functionality directly against the sandbox, since auth stuff needed to exists beforehand.

**Review by ltaziri — APPROVED** [review]

**Pull request merged**